scout/run.py

```diff
@@ -6869,12 +6882,29 @@ def sum_energy_data(self, reg, bldg, vint, adopt_scheme, prior_yr_rmv):
             "efficient": {yr: 0 for yr in self.handyvars.aeo_years}
         }
 
-        # Loop through all measures and sum energy across given regions, building types/vintages
-        # of focus for the current code/BPS policy
-        for m in [m_s for m_s in self.measures if (
-                reg in m_s.climate_zone and bldg in m_s.bldg_type and vint in m_s.structure_type)]:
-            # Shorthand for measure energy breakout data
-            brk_dat = m.markets[adopt_scheme]["competed"]["mseg_out_break"]["energy"]
+        # Input measure set to sum (exclusive of any code/BPS measures that have been added on top
+        # of the input measure set). Ensure measures apply to current region, building type, and
+        # building vintage being processed
+        start_meas_to_sum = [m_s for m_s in self.measures if (
+            any([x in self.handyvars.out_break_czones[reg] for x in m_s.climate_zone]) and
+            any([x in self.handyvars.out_break_bldgtypes[bldg] for x in m_s.bldg_type]) and
+            vint in m_s.structure_type)]
+
+        # Add any code/BPS measures to the original measure set to get a full picture of energy use,
+        # inclusive of the energy effects of any code/BPS policies that require reductions of
+        # onsite emissions (and fuel switching to electric equipment, which may be more efficient).
+        # These onsite policies will have already been reflected.
+        meas_to_sum = start_meas_to_sum + code_bps_meas_to_sum
+
+        # Loop through all measures (including previously-assessed code/BPS policies) and sum energy
+        # across given regions, building types/vintages of focus for the current code/BPS policy
+        for m in meas_to_sum:
+            # Shorthand for measure energy breakout data; handle code/BPS measure data, which will
+            # not be broken out by 'uncompeted' and 'competed' keys (see Codes_BPS_Measure object)
+            try:
+                brk_dat = m.markets[adopt_scheme]["competed"]["mseg_out_break"]["energy"]
+            except KeyError:
```

Comment:
We should also check the impact of try/except vs .get in the profilers. It should matter with memory but may with runtime. I prefer the .get syntax from an intent perspective but the comments make that just a preference.